File: LeetCode/reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        s = s.strip()
        words = s.split(" ")
        finalWords = []
        for word in words:
            if word != "":
                finalWords.append(word)
        finalWords.reverse()
        return " ".join(finalWords)

File: LeetCode/reverse_words_in_a_string.java
import java.util.StringTokenizer;
class Solution {
    public String reverseWords(String s) {
        StringTokenizer st = new StringTokenizer(s);
        StringBuffer sb = new StringBuffer();
        while(st.hasMoreTokens()){
            String word = st.nextToken();
            for(int iter = word.length()-1; iter>= 0; iter--){
                sb.append(word.charAt(iter));
            }
            if(st.hasMoreTokens()){
                sb.append(" ");
            }
        }
        return sb.toString();
    }
}